website/templates/deviceactions.html

```diff
@@ -1,103 +1,68 @@
+        <br>
+        <li><input type="radio" name="motionsensor" value="OFF"> Motion Sensor is OFF</li>
+        <br>
+    </form>
+    <hr>
+    <br>
+    <h3> Kitchen Motion Triggers: </h3>
+    <hr>
+    {% for trigger in triggers %}
+        <div class="dropdown">
+            <button type="button" class="btn btn-primary dropdown-toggle" data-toggle="dropdown"> {{ trigger }}
+                <span class="caret"></span>
+            </button>
+            <ul class="dropdown-menu" role="menu">
+                {% for actor in actors %}
+                    <li><a href="#lightswitch" data-toggle="modal"> {{ actor }} </a></li>
+                {% endfor %}
+                <li class="divider"></li>
+                <li><a href="#"> Send E-mail </a></li>
+            </ul>
         </div>
-    </div>
-</div>
-<br>
-<br>
-<div class="dropdown">
-    <button type="button" class="btn btn-primary dropdown-toggle" data-toggle="dropdown"> When Motion is Detected
-        <span class="caret"></span>
-    </button>
-    <ul class="dropdown-menu" role="menu">
-        <li><a href="#lightswitch" data-toggle="modal"> Light Switch </a></li>
-        <li><a href="#"> Thermostat </a></li>
-        <li class="divider"></li>
-        <li><a href="#"> Send E-mail </a></li>
-    </ul>
-</div>
-<div class="modal fade" id="lightswitch" data-target="#lightswitch">
-    <div class="modal-dialog">
-        <div class="modal-content">
-            <div class="modal-header orange">
-                <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span
-                        aria-hidden="true">&times;</span>
-                </button>
-                <h4 class="modal-title"><strong></strong> Light Switch</h4>
-                <ul>
-                    <li>
-                        <label class="radio-inline"> <input type="radio" name="MotionSensor" id="MotionSensor"
-                                                            value="ON"> Turn on </label>
-                    </li>
-                    <li>
-                        <label class="radio-inline"> <input type="radio" name="MotionSensor" id="MotionSensor"
-                                                            value="OFF">
-                            Turn off </label>
-                    </li>
-                    <li><label class="radio-inline"> <input type="radio" name="MotionSensor" id="MotionSensor"
-                                                            value="OFF">
-                        No Action </label>
-                    </li>
+        <br>
+    {% endfor %}
 
-                </ul>
-                <div class="modal-footer">
-                    <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
-                    <button type="button" class="btn btn-primary"> Done</button>
+    <div class="modal fade" id="lightswitch" data-target="#lightswitch">
```

Comment:
How do you want to distinguish between different actors? 
Suggestion: you could add modals for every type of sensor, or you could load the modal dynamically whenever an actor is selected. (Have a look at e.g. http://api.jquery.com/load/). 
What do you think?